awsapilib/console/console.py

```diff
@@ -499,8 +502,12 @@ def _get_root_console_redirect(self, email, password, session, mfa_serial=None):
                               f'with status code: {response.status_code}')
         oidc = self._get_oidc_info(response.history[0].headers.get('Location'))
         response = self._resolve_account_type_response(email, session=session)
+        captcha_url = response.json().get('properties', {}).get('CaptchaURL')
+        captcha_status_token = response.json().get('properties', {}).get('captchaStatusToken')
+
         if any([not response.ok,
-                response.json().get('properties', {}).get('captchaStatusToken') is None]):
+                all([captcha_url is not None,
```

Comment:
Should this also be `any` instead of `all`?

Reply:
No, because if there is no captcha url, the `captchaStatusToken` is going to be empty, and in this case it would fail!
So,
```python
In [1]: captcha_url = None

In [2]: captcha_status_token = None

In [3]: any([captcha_url is not None, captcha_status_token is None])
Out[3]: True

In [4]: all([captcha_url is not None, captcha_status_token is None])
Out[4]: False
```

Reply:
you only want this to be True, if there is an URL and the token is empty!

Reply:
Gotcha!